idx/memory/tag_query.go

```diff
@@ -50,36 +72,61 @@ func (a KvReByCost) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
 func (a KvReByCost) Less(i, j int) bool { return a[i].cost < a[j].cost }
 
 type TagQuery struct {
```

Comment:
let's document what this type is for. how about something like: 
> TagQuery runs a set of pattern or string matches on tag keys and values against the index. It is executed via: 
> Run() which returns a set of matching MetricID's
> RunGetTags() which returns the set of tag values found amongst the matching MetricID's

I'm not sure if this is entirely correct though. does the latter return tag keys or values, or could it be either one? Do both Run functions all support the same set of expressions?

Reply:
the latter does only return slices of strings (tags). they support the same query expressions